x/outline-connectivity-app/shared_backend/index.ts

```diff
@@ -0,0 +1,25 @@
+import { ConnectivityTestInput, ConnectivityTestResult } from "./generated/types"
+
+export function from(backend: Invokable) {
+  return {
+    connectivityTest: invocationFactory<ConnectivityTestInput, ConnectivityTestResult[]>(backend, "ConnectivityTest")
+  }
+}
+
+/* INFRASTRUCTURE (to be moved) */
+
+export interface Invokable {
+  Invoke(parameters: { method: string; input: string; }): Promise<{ result: string; errors: string[] }>
```

Comment:
The output error should be a single error, not a list. It's hard to reason about a list. It's definitely not a common practice.

The error would need to be a struct, with a message and an optional cause that we can unwrap if needed.

That will nicely map to the JS Error.